docs/usage/secrets_engines/database.rst

```diff
@@ -0,0 +1,251 @@
+Database
+==============
+
+.. note::
+    Every method under the :py:attr:`Database class attribute<hvac.api.secrets_engines.Database>` includes a `mount_point` parameter that can be used to address the Database secret engine under a custom mount path. E.g., If enabling the Database secret engine using Vault's CLI commands via `vault secrets enable -path=my-database database`", the `mount_point` parameter in :py:meth:`hvac.api.secrets_engines.Database()` methods would be set to "my-database".
```

Comment:
```suggestion
    Every method under the :py:attr:`Database class<hvac.api.secrets_engines.Database>` includes a ``mount_point`` parameter that can be used to address the Database secret engine under a custom mount path. E.g., If enabling the Database secret engine using Vault's CLI commands via ``vault secrets enable -path=my-database database``, the ``mount_point`` parameter in :py:meth:`hvac.api.secrets_engines.Database()` methods would be set to ``my-database``.
```
- to use code type formatting (`like this`) in RST we need to use double backticks <code>``</code> instead of single <code>`</code> like we would in MD. 
  - I know other places in our docs are not doing this right now so you did the right thing to mirror that convention. I'd like to see those updated at some point too but it doesn't have to be all at once so I figured let's start here since it's a new page.

nits:
- removed the word "attribute" from the class name
- removed extra `"` character in the `vault` command